pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -232,3 +205,39 @@ func UpdateNodeOwnerReferences(nodeClaim *v1beta1.NodeClaim, node *v1.Node) *v1.
 	})
 	return node
 }
+
+// TerminationHandler is a helper function that takes a v1beta1.NodeClaim and calls cloudProvider.Delete() if status condition
+// on nodeClaim is not terminating. If it is terminating then it will check if the kubelet process is still running. It will
+// return an error and a boolean that indicates if the reconciliation should be re-enqueued.
+func TerminationHandler(ctx context.Context, c client.Client, nodeClaim *v1beta1.NodeClaim, node *v1.Node, cloudProvider cloudprovider.CloudProvider) (bool, error) {
+	// Check if the status condition on nodeClaim is Terminating
+	terminationStatus := nodeClaim.StatusConditions().GetCondition(v1beta1.Terminating)
+	if terminationStatus.IsUnknown() || terminationStatus.IsFalse() {
+		// If not then call Delete on cloudProvider to trigger termination and always requeue reconciliation
+		if err := cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
```

Comment:
When I first implemented this, I thought it would be better to have terminating status set on nodeClaim even if the underlying instance is gone so that there is some kind of uniformity in the status condition behavior. It will simply take less time to delete if the instance is already gone and atleast customer would know that their nodeClaim will be gone once that status is set. 
However, we can argue on if that status should just indicate termination of underlying instance. And if that's all we want to convey, then it would make sense to bail out. But then instead of calling it `Terminating` we should call it something else.